lib/NormalModule.js

```diff
@@ -442,9 +441,37 @@ class NormalModule extends Module {
 				return callback();
 			}
 
+			const getLoaders = options => {
+				const module =
+					options && options.constructor === Object ? options.module || {} : {};
+				const rules = module.rules || [];
+				if (!rules.length) {
+					return [];
+				}
+				let arr = [];
+				const getLoader = rule => rule.loader || rule.use || "";
+				rules.forEach(rule => {
+					if (Array.isArray(rule.use)) {
+						rule.use.forEach(r => {
+							arr.push(getLoader(r));
+						});
+					} else {
+						arr.push(getLoader(rule));
+					}
+				});
+				return arr;
+			};
+
 			const handleParseError = e => {
 				const source = this._source.source();
-				const error = new ModuleParseError(this, source, e);
+				const loaders = getLoaders(options);
```

Comment:
You can get the used loaders from `this.loaders`.

Reply:
Yeah, that's what I tried using it at first but I noticed `this.loaders` was empty by the time `this.doBuild` is called. 

<img width="1216" alt="Screen Shot 2019-05-20 at 8 46 30 AM" src="https://user-images.githubusercontent.com/6132051/58034657-ed597480-7adb-11e9-95c6-5ce8aee2e4a6.png">

Reply:
That shouldn't be the case. `this.loaders` is assigned in the constructor and never changed again. It's passed to `runLoaders`, so technically this should be fine.

Reply:
Yeah, I've noticed that it depends on the invocation of `doBuild`. It seems like on the first run `this.loaders`is populated but right before the error is thrown they're discarded somehow. I also tried getting the loaders from `this.issuer.loaders`, which seems like it has the data by the time `handleParseError` is called. Should I retrieve it from there?